homeworks/08-functional-reactive-programming/src/App.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import {
+  combineLatest, from, interval, of, Subject,
+} from 'rxjs';
+import { useObservableState } from 'observable-hooks';
+import { map, scan } from 'rxjs/operators';
+import './App.css';
+
+import { Grid } from './components/Grid';
+
+import { generateGrid, getRandomTarget } from './utils';
+
+import { CellContent } from './types';
+
+export const App: React.FC = () => {
+  const [goodShots, setGoodShots] = useState<number>(0);
+
+  const shotTarget = new Subject<CellContent>();
+  const shotTarget$ = shotTarget.asObservable();
+
+  const targetIndex$ = interval(1000).pipe(map(() => getRandomTarget()));
+
+  const grid$ = of(generateGrid());
+
+  const cellsWithTarget$ = combineLatest([targetIndex$, grid$])
+    .pipe(
+      map(([targetIndex, grid]) => {
+        const newGrid = [...grid];
+        newGrid[targetIndex].content = CellContent.CAT;
+        return newGrid;
+      }),
+    );
+
+  const goodShots$ = from(shotTarget$)
+    .pipe(
+      map((target) => (target === CellContent.CAT ? 1 : 0)),
+      scan((acc, count) => acc + count, 0),
+    );
+
+  const cellsWithTarget = useObservableState(cellsWithTarget$, []);
+
+  useEffect(() => {
+    const sub = goodShots$.subscribe((v) => {
+      setGoodShots(v);
+    });
+    return sub.unsubscribe;
+  }, []);
+
+  const handleClick = useCallback((target: CellContent) => {
+    shotTarget.next(target);
+  }, []);
```

Comment:
Again dependency array is empty.

Reply:
Added game grid as a dependency. 

> There is a bug: each click on a cat postpones re-creating a cat for 1 second. So I can easy achieve a case when a cat always stays in one window.

And this bug fixed during refactoring :)